mito-ai/src/Extensions/AppDeploy/requirementsUtils.tsx

```diff
@@ -32,23 +34,29 @@ if not os.path.exists(app_py_path):
     print(f"Error: {app_py_filename} not found at {app_py_path}")
     exit(1)
 
-# Make sure pipreqs is installed. Then
-# 1. Create a requirements.in file
-# 2. From the requirements.in file, generate the requirements.txt file with the canonical PyPI name of the packages
-# and the versions as they exist on the user's terminal
 try:
-    # Run pipreqs on the directory containing app.py
-    notebook_dir = os.path.dirname(app_py_path)
-    generate_req_in_file = subprocess.run(
-        ['pipreqs', '--encoding=utf-8', '--savepath', 'requirements.in', '--force', notebook_dir],
-        capture_output=True, 
-        text=True
-    )
+    # Create a temporary directory and copy the app file into it
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmp_app_path = os.path.join(tmpdir, app_py_filename)
+        shutil.copy(app_py_path, tmp_app_path)
+
+        # Run pipreqs on the temporary directory
+        generate_req_in_file = subprocess.run(
+            [
+                'pipreqs',
+                '--encoding=utf-8',
+                '--savepath', os.path.join(os.getcwd(), 'requirements.in'),
+                '--force',
+                tmpdir
+            ],
+            capture_output=True,
+            text=True
+        )
 
     print("Log: ", generate_req_in_file.stderr)
 
     # Read requirements.in and process each line
-    requirements_in_path = os.path.join(notebook_dir, 'requirements.in')
+    requirements_in_path = os.path.join(os.getcwd(), 'requirements.in')
```

Comment:
Why do we switch this from notebook_dir to getcwd()? most of the time those would be the same thing I think, but we caught some bugs when we standardized things and used the same approach across everything. So does it make sense to go back to `os.path.dirname(app_py_path)`?